contracts/strategies/PenguinIglooStrategyV2.sol

```diff
@@ -0,0 +1,247 @@
+    require(unclaimedRewards >= MIN_TOKENS_TO_REINVEST, "PenguinIglooStrategyV2::reinvest");
+    _reinvest(unclaimedRewards);
+  }
+
+  /**
+    * @notice Reinvest rewards from staking contract to deposit tokens
+    * @dev Reverts if the expected amount of tokens are not returned from `stakingContract`
+    * @param unclaimedRewards deposit tokens to reinvest
+    */
+  function _reinvest(uint unclaimedRewards) private {
+    stakingContract.harvest(PID, address(this));
+    uint amount = _convertRewardIntoWAVAX(unclaimedRewards);
+    uint devFee = amount.mul(DEV_FEE_BIPS).div(BIPS_DIVISOR);
+        if (devFee > 0) {
+            _safeTransfer(address(WAVAX), devAddr, devFee);
+        }
+
+        uint adminFee = amount.mul(ADMIN_FEE_BIPS).div(BIPS_DIVISOR);
+        if (adminFee > 0) {
+            _safeTransfer(address(WAVAX), owner(), adminFee);
+        }
+
+        uint reinvestFee = amount.mul(REINVEST_REWARD_BIPS).div(BIPS_DIVISOR);
+        if (reinvestFee > 0) {
+            _safeTransfer(address(WAVAX), msg.sender, reinvestFee);
+        }
+
+        uint depositTokenAmount = DexLibrary.convertRewardTokensToDepositTokens(
+            amount.sub(devFee).sub(adminFee).sub(reinvestFee),
+            address(WAVAX),
+            address(depositToken),
+            swapPairToken0,
+            swapPairToken1
+        );
+    _stakeDepositTokens(depositTokenAmount);
+    emit Reinvest(totalDeposits(), totalSupply);
+  }
+    
+  function _stakeDepositTokens(uint amount) private {
+    require(amount > 0, "PenguinIglooStrategyV2::_stakeDepositTokens");
+    stakingContract.deposit(PID, amount, address(this));
+  }
+
+  function checkReward() public override view returns (uint) {
+    uint pendingReward = stakingContract.pendingPEFI(PID, address(this));
+    uint contractBalance = rewardToken.balanceOf(address(this));
+    return pendingReward.add(contractBalance);
+  }
+
+  function _convertRewardIntoWAVAX(uint pendingReward) private returns (uint) {
+        return DexLibrary.swap(
+            pendingReward,
+            address(rewardToken), address(WAVAX),
+            swapPairWAVAXPEFI
+        );
+    }
+
+  /**
+     * @notice Safely transfer using an anonymosu ERC20 token
+     * @dev Requires token to return true on transfer
+     * @param token address
+     * @param to recipient address
+     * @param value amount
+     */
+    function _safeTransfer(address token, address to, uint256 value) private {
+        require(IERC20(token).transfer(to, value), 'PenguinIglooStrategyV2::_safeTransfer');
+    }
+
+  /**
+   * @notice Estimate recoverable balance after withdraw fee
+   * @return deposit tokens after withdraw fee
+   */
+  function estimateDeployedBalance() external override view returns (uint) {
+    (uint depositBalance, ) = stakingContract.userInfo(PID, address(this));
+    (,,,,,, uint withdrawFeeBP,,) = stakingContract.poolInfo(PID);
+    uint withdrawFee = depositBalance.mul(withdrawFeeBP).div(BIPS_DIVISOR);
```

Comment:
do we really want to account for withdrawFee here? I mean, I'll take a cut after withdraw, I know this, but if this computation is done here then someone might double do it no?

Reply:
This is an external view function which is not being used from inside the contract so I am not sure of the scenario where someone use estimateDeployedBalance of the whole contract in their computation rather than just user based calculation?

Reply:
I think front-end uses to compute TVL, could also be using to compute how much depositToken you have on the strategy right now(the front end shows how much you deposited at first and how much you have right now), so it's kinda of a key piece for the protocol

Reply:
Removed the withdrawfee inclusion